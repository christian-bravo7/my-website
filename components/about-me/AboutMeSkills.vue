<template>
  <section class="section min-screen skills-section">
    <div class="skills-section__container container">
      <h2 class="has-text-centered skills-section__title">
        {{ $t('me.skills-label') }}
      </h2>
      <div class="skills-section__collection">
        <SkillIcon name="HTML5">
          <HtmlIcon />
        </SkillIcon>
        <SkillIcon name="CSS3">
          <CssIcon />
        </SkillIcon>
        <SkillIcon name="SASS">
          <SassIcon />
        </SkillIcon>
        <SkillIcon name="Nuxt.js">
          <NuxtIcon />
        </SkillIcon>
        <SkillIcon name="React.js">
          <ReactIcon />
        </SkillIcon>
        <SkillIcon name="Vue.js">
          <VueIcon />
        </SkillIcon>
        <SkillIcon name="Typescript">
          <TypescriptIcon />
        </SkillIcon>
        <SkillIcon name="Javascript">
          <JavascriptIcon />
        </SkillIcon>
        <SkillIcon name="Node.js">
          <NodeIcon />
        </SkillIcon>
        <SkillIcon name="GraphQL">
          <GraphqlIcon />
        </SkillIcon>
        <SkillIcon name="Git">
          <GitIcon />
        </SkillIcon>
        <SkillIcon name="Jest">
          <JestIcon />
        </SkillIcon>
      </div>
    </div>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import CssIcon from '~/assets/img/css3.svg?inline';
import JavascriptIcon from '~/assets/img/javascript.svg?inline';
import NuxtIcon from '~/assets/img/nuxt-dot-js.svg?inline';
import ReactIcon from '~/assets/img/react.svg?inline';
import VueIcon from '~/assets/img/vue-dot-js.svg?inline';
import TypescriptIcon from '~/assets/img/typescript.svg?inline';
import NodeIcon from '~/assets/img/node-dot-js.svg?inline';
import GraphqlIcon from '~/assets/img/graphql.svg?inline';
import GitIcon from '~/assets/img/git.svg?inline';
import SassIcon from '~/assets/img/sass.svg?inline';
import JestIcon from '~/assets/img/jest.svg?inline';
import HtmlIcon from '~/assets/img/html5.svg?inline';

@Component({
  components: {
    CssIcon,
    JavascriptIcon,
    NuxtIcon,
    ReactIcon,
    VueIcon,
    TypescriptIcon,
    GraphqlIcon,
    GitIcon,
    SassIcon,
    JestIcon,
    HtmlIcon,
    NodeIcon,
  },
})
export default class AboutMeSkills extends Vue {}

</script>

<style lang="scss" scoped>
.skills-section {
  display: flex;
  align-items: center;

  &__container {
    display: grid;
    grid-template-columns: 1fr;
  }

  &__title {
    margin-bottom: rem(24);
    color: var(--text-light);
    font-weight: bold;
    font-size: rem(182);
    word-break: break-word;
  }

  &__collection {
    display: grid;
    grid-gap: rem(25);
    grid-template-columns: repeat(auto-fill, minmax(150px, 1fr));
    justify-items: center;
    padding: rem(24) 0;
  }
}

.home {
  &__content {
    padding: rem(92) rem(24);
    background-color: var(--background);

    &--secondary {
      background-color: var(--text-light);
    }

    &--tertiary {
      background-color: var(--background-footer);
    }
  }
}

@include mobile {
  .skills-section {
    &__title {
      font-size: rem(70);
    }

    &__collection {
      grid-gap: rem(10) rem(5);
    }
  }
}
</style>
