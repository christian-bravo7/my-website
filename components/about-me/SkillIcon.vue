<template>
  <div class="flex flex-col group items-center justify-center w-full p-2 skill-icon hover:shadow-lg transition-shadow duration-150">
    <span class="w-16 text-pink-800 dark:text-blue-500">
      <slot name="default" />
    </span>
    <span class="mt-2 text-gray-300 dark:text-blue-700 group-hover:text-pink-800 dark-group-hover:text-blue-500 text-base md:text-lg font-bold transition-colors duration-150">
      {{ name }}
    </span>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';

@Component
export default class SkillIcon extends Vue {
  @Prop({ type: String, required: true })
  readonly name!: string;
}
</script>

<style lang="scss" scoped>
.skill-icon {
  svg {
    @apply fill-current;
  }
}
</style>
