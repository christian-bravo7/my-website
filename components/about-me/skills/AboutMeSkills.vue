<template>
  <section class="section-container">
    <div class="container | flex | my-0 mx-auto">
      <div class="flex flex-col | w-full">
        <h2 class="mb-8 | text-center break-all | section-title about-me-skills__title">
          {{ $t('me.skills-label') }}
        </h2>
        <div class="flex justify-center items-center | flex-1">
          <div class="flex flex-wrap justify-center | max-w-screen-lg">
            <SkillIcon
              v-for="({ name, component }, index) in skills"
              :key="index"
              :name="name"
              class="m-2"
            >
              <Component :is="component" />
            </SkillIcon>
          </div>
        </div>
      </div>
    </div>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

import CssIcon from '~/assets/img/css3.svg?inline';
import JavascriptIcon from '~/assets/img/javascript.svg?inline';
import NuxtIcon from '~/assets/img/nuxt-dot-js.svg?inline';
import ReactIcon from '~/assets/img/react.svg?inline';
import VueIcon from '~/assets/img/vue-dot-js.svg?inline';
import TypescriptIcon from '~/assets/img/typescript.svg?inline';
import NodeIcon from '~/assets/img/node-dot-js.svg?inline';
import GitIcon from '~/assets/img/git.svg?inline';
import SassIcon from '~/assets/img/sass.svg?inline';
import JestIcon from '~/assets/img/jest.svg?inline';
import HtmlIcon from '~/assets/img/html5.svg?inline';

@Component({
  components: {
    CssIcon,
    JavascriptIcon,
    NuxtIcon,
    ReactIcon,
    VueIcon,
    TypescriptIcon,
    GitIcon,
    SassIcon,
    JestIcon,
    HtmlIcon,
    NodeIcon,
  },
})
export default class AboutMeSkills extends Vue {
  skills: Array<{ name: string, component: any }> = [
    { name: 'Javascript', component: JavascriptIcon },
    { name: 'Typescript', component: TypescriptIcon },
    { name: 'NuxtJS', component: NuxtIcon },
    { name: 'VueJS', component: VueIcon },
    { name: 'ReactJS', component: ReactIcon },
    { name: 'NodeJS', component: NodeIcon },
    { name: 'Git', component: GitIcon },
    { name: 'Jest', component: JestIcon },
    { name: 'SASS', component: SassIcon },
    { name: 'CSS3', component: CssIcon },
    { name: 'HTML5', component: HtmlIcon },
  ]
}
</script>

<style lang="scss" scoped>
.about-me-skills {
  &__title {
    @extend .gradient-one;
  }
}

html.dark-mode {
  .about-me-skills {
    &__title {
      @extend .gradient-two;
    }
  }
}
</style>
