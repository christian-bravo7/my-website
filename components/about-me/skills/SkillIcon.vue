<template>
  <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-blue-700 | rounded-xl shadow-md hover:shadow-xl | select-none | group transition-shadow duration-300 | skill-icon">
    <span class="flex justify-center items-center flex-1 | w-16">
      <slot name="default" />
    </span>
    <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-blue-500 | text-base md:text-lg | font-bold | transition-colors duration-300">
      {{ name }}
    </span>
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';

@Component
export default class SkillIcon extends Vue {
  @Prop({ type: String, required: true })
  readonly name!: string;
}
</script>

<style lang="scss" scoped>
.skill-icon {
  min-width: 140px;

  svg {
    @apply fill-current;
  }
}
</style>
