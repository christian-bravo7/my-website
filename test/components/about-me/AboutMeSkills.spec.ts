import { mount } from '@vue/test-utils';

import AboutMeSkills from '@/components/about-me/AboutMeSkills.vue';
import SkillIcon from '@/components/about-me/SkillIcon.vue';

describe('AboutMeSkills component', () => {
  it('should render correctly', () => {
    const wrapper = mount(AboutMeSkills, {
      components: {
        SkillIcon,
      },
      mocks: {
        $t: jest.fn(msg => msg),
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display a title', () => {
    const wrapper = mount(AboutMeSkills, {
      components: {
        SkillIcon,
      },
      mocks: {
        $t: jest.fn(msg => msg),
      },
    });

    const title = wrapper.find('h2');

    expect(title.text()).toBe('me.skills-label');
  });

  it('should render multiple SkillIcon components', () => {
    const wrapper = mount(AboutMeSkills, {
      components: {
        SkillIcon,
      },
      mocks: {
        $t: jest.fn(msg => msg),
      },
    });

    const icons = wrapper.findAllComponents(SkillIcon);

    expect(icons.length).toBe(12);
  });
});
