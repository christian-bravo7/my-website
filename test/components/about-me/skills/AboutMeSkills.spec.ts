import { mount } from '@vue/test-utils';

import AboutMeSkills from '@/components/about-me/skills/AboutMeSkills.vue';
import SkillIcon from '@/components/about-me/skills/SkillIcon.vue';

describe('AboutMeSkills component', () => {
  it('should render correctly', () => {
    const wrapper = mount(AboutMeSkills, {
      components: {
        SkillIcon,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display a title', () => {
    const wrapper = mount(AboutMeSkills, {
      components: {
        SkillIcon,
      },
    });

    const title = wrapper.find('h2');

    expect(title.text()).toBe('me.skills-label');
  });

  it('should render multiple SkillIcon components', () => {
    const wrapper = mount(AboutMeSkills, {
      components: {
        SkillIcon,
      },
    });

    const icons = wrapper.findAllComponents(SkillIcon);

    expect(icons.length).toBe(11);
  });
});
