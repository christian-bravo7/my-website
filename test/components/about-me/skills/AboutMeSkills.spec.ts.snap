// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMeSkills component should render correctly 1`] = `
<section class="section-container">
  <div class="container | flex | my-0 mx-auto">
    <div class="flex flex-col | w-full">
      <h2 class="mb-8 | text-center break-all | section-title about-me-skills__title">
        me.skills-label
      </h2>
      <div class="flex justify-center items-center | flex-1">
        <div class="flex flex-wrap justify-center | max-w-screen-lg">
          <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-transparent dark-hover:bg-blue-700 dark-hover:bg-opacity-25 | rounded-xl shadow-md hover:shadow-xl border border-transparent dark:border-blue-300 | select-none | group transition-all duration-300 | skill-icon m-2"><span class="flex justify-center items-center flex-1 | w-16"><svg></svg></span> <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-white | text-base md:text-lg | font-bold leading-none | transition-colors duration-300">
    Javascript
  </span></div>
          <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-transparent dark-hover:bg-blue-700 dark-hover:bg-opacity-25 | rounded-xl shadow-md hover:shadow-xl border border-transparent dark:border-blue-300 | select-none | group transition-all duration-300 | skill-icon m-2"><span class="flex justify-center items-center flex-1 | w-16"><svg></svg></span> <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-white | text-base md:text-lg | font-bold leading-none | transition-colors duration-300">
    Typescript
  </span></div>
          <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-transparent dark-hover:bg-blue-700 dark-hover:bg-opacity-25 | rounded-xl shadow-md hover:shadow-xl border border-transparent dark:border-blue-300 | select-none | group transition-all duration-300 | skill-icon m-2"><span class="flex justify-center items-center flex-1 | w-16"><svg></svg></span> <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-white | text-base md:text-lg | font-bold leading-none | transition-colors duration-300">
    NuxtJS
  </span></div>
          <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-transparent dark-hover:bg-blue-700 dark-hover:bg-opacity-25 | rounded-xl shadow-md hover:shadow-xl border border-transparent dark:border-blue-300 | select-none | group transition-all duration-300 | skill-icon m-2"><span class="flex justify-center items-center flex-1 | w-16"><svg></svg></span> <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-white | text-base md:text-lg | font-bold leading-none | transition-colors duration-300">
    VueJS
  </span></div>
          <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-transparent dark-hover:bg-blue-700 dark-hover:bg-opacity-25 | rounded-xl shadow-md hover:shadow-xl border border-transparent dark:border-blue-300 | select-none | group transition-all duration-300 | skill-icon m-2"><span class="flex justify-center items-center flex-1 | w-16"><svg></svg></span> <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-white | text-base md:text-lg | font-bold leading-none | transition-colors duration-300">
    ReactJS
  </span></div>
          <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-transparent dark-hover:bg-blue-700 dark-hover:bg-opacity-25 | rounded-xl shadow-md hover:shadow-xl border border-transparent dark:border-blue-300 | select-none | group transition-all duration-300 | skill-icon m-2"><span class="flex justify-center items-center flex-1 | w-16"><svg></svg></span> <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-white | text-base md:text-lg | font-bold leading-none | transition-colors duration-300">
    NodeJS
  </span></div>
          <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-transparent dark-hover:bg-blue-700 dark-hover:bg-opacity-25 | rounded-xl shadow-md hover:shadow-xl border border-transparent dark:border-blue-300 | select-none | group transition-all duration-300 | skill-icon m-2"><span class="flex justify-center items-center flex-1 | w-16"><svg></svg></span> <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-white | text-base md:text-lg | font-bold leading-none | transition-colors duration-300">
    Git
  </span></div>
          <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-transparent dark-hover:bg-blue-700 dark-hover:bg-opacity-25 | rounded-xl shadow-md hover:shadow-xl border border-transparent dark:border-blue-300 | select-none | group transition-all duration-300 | skill-icon m-2"><span class="flex justify-center items-center flex-1 | w-16"><svg></svg></span> <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-white | text-base md:text-lg | font-bold leading-none | transition-colors duration-300">
    Jest
  </span></div>
          <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-transparent dark-hover:bg-blue-700 dark-hover:bg-opacity-25 | rounded-xl shadow-md hover:shadow-xl border border-transparent dark:border-blue-300 | select-none | group transition-all duration-300 | skill-icon m-2"><span class="flex justify-center items-center flex-1 | w-16"><svg></svg></span> <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-white | text-base md:text-lg | font-bold leading-none | transition-colors duration-300">
    SASS
  </span></div>
          <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-transparent dark-hover:bg-blue-700 dark-hover:bg-opacity-25 | rounded-xl shadow-md hover:shadow-xl border border-transparent dark:border-blue-300 | select-none | group transition-all duration-300 | skill-icon m-2"><span class="flex justify-center items-center flex-1 | w-16"><svg></svg></span> <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-white | text-base md:text-lg | font-bold leading-none | transition-colors duration-300">
    CSS3
  </span></div>
          <div class="flex flex-col items-center justify-between | p-2 | bg-gray-100 dark:bg-transparent dark-hover:bg-blue-700 dark-hover:bg-opacity-25 | rounded-xl shadow-md hover:shadow-xl border border-transparent dark:border-blue-300 | select-none | group transition-all duration-300 | skill-icon m-2"><span class="flex justify-center items-center flex-1 | w-16"><svg></svg></span> <span class="mt-2 | text-black dark:text-white group-hover:text-pink-500 dark-group-hover:text-white | text-base md:text-lg | font-bold leading-none | transition-colors duration-300">
    HTML5
  </span></div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
